File: 1338-reduce-array-size-to-the-half/1338-reduce-array-size-to-the-half.cpp
class Solution {
public:
    int minSetSize(vector<int>& arr) {
       unordered_map<int,int> mp;
        priority_queue<int> pq;
    
    
        int count =0 ; 
        int res=0;

        for(int i=0  ;i<arr.size();i++)
        {
            mp[arr[i]]++;
        }
        for(auto c : mp)
        {
            pq.push(c.second);
        }

        while(count<arr.size()/2)
        {
            count += pq.top();
            pq.pop();

            res++;
        }
    
     return res;
    }
    
};

File: 1338-reduce-array-size-to-the-half/1338-reduce-array-size-to-the-half.java
class Solution {
    public int minSetSize(int[] arr) {
        Map<Integer, Integer> mp = new HashMap<>();
        PriorityQueue<Integer> pq = new PriorityQueue<>(Comparator.reverseOrder());
        for (int num : arr) mp.put(num, mp.getOrDefault(num, 0) + 1);
        for (int value : mp.values()) pq.offer(value);

        int size = arr.length; int result = 0;
        while (size > arr.length / 2) {
            size -= pq.poll();
            result++;
        }

        return result;
    }
}